tsdb/db_test.go

```diff
@@ -155,6 +155,65 @@ func TestDataAvailableOnlyAfterCommit(t *testing.T) {
 	testutil.Equals(t, map[string][]tsdbutil.Sample{`{foo="bar"}`: {sample{t: 0, v: 0}}}, seriesSet)
 }
 
+// TestNoPanicAfterWALCorrutpion ensures that querying the db after a WAL corruption doesn't cause a panic.
+// https://github.com/prometheus/prometheus/issues/7548
+func TestNoPanicAfterWALCorrutpion(t *testing.T) {
+	db, closeFn := openTestDB(t, &Options{WALSegmentSize: 32 * 1024}, nil)
+	defer func() {
+		closeFn()
+	}()
+
+	// Append untill we have at least one mmaped chunk.
+	var expSamples []tsdbutil.Sample
+	var maxt int64
+	{
+		app := db.Appender()
```

Comment:
Should we not have this inside the `for` loop? We have a change incoming which will make Commit/Rollback on the same appender invalid.

Reply:
hm ,that is strange, with a single appended it creates only a single wal file, investigating